enable Xdebug with docker build stage and dc override

lwillems opened the pull request:
Hello

To enable Xdebug just add a new stage to DockerFile and load override.
Original credits go to @dunglas with [docker sf5 repo ](https://github.com/dunglas/symfony-docker)

Hope it could be useful.

Regards

lwillems committed:
enable Xdebug with docker build stage and dc override

lchrusciel commented:
Hey Laurent,
thansk for your proposal. However, the build is red. I've restarted it.

lwillems commented:
@lchrusciel 
Build seems fail on php memory limit issue ?

lchrusciel commented:
Hey Laurent, we need to wait for the green build of #436. After merging and rebase, we can move on.

lwillems committed:
fix typo error

lwillems commented:
@lchrusciel i have fixed a typo issue breaking build with overidde, seems #436 merged ?
Regards

lchrusciel commented:
Hey Laurent, the current issue is related to Docker itself: https://travis-ci.com/github/Sylius/Sylius-Standard/jobs/423497979#L2986

Have you tested the current build on your machine? Do you think, you could fix our build?

lwillems committed:
fix docker build

lwillems commented:
> Hey Laurent, the current issue is related to Docker itself: https://travis-ci.com/github/Sylius/Sylius-Standard/jobs/423497979#L2986
> 
> Have you tested the current build on your machine? Do you think, you could fix our build?

Hi @lchrusciel 
After investigations
I think build fails for several reason but the most important one is composer version
Prestissimo package is locked to composer 1 and in your build you are using composer:latest (now is version 2 by default).
I have fixed it with latest version on docker hub : 2.0.4 (latest should do the job but locally it always pull 1.10 on my laptop), also upgraded to php 7.4 as suggested by Dunglas in another PR.
Also there is differences between lock and json file, i have updated composer.lock file but there is conflict

Can you resolve conflict with my file (up to date) and relaunch build with latest changes ? it should work, on my local machine it's works.

Regards

lchrusciel commented:
Done in separe PR, as I was not able to add something to your branch

lchrusciel commented:
Hey Laurent,

it has been way too long since you received some feedback from us. Is there chance, to rebase this PR?

Ferror commented:
@lwillems Thank you for your PR!

The PR contains also changed not related to Docker and Xdebug. We are more than happy to merge it, but without additional changes in symfony.lock, composer.lock and some of the Dockerfile changes.

Ferror commented:
Created the development stage in: https://github.com/Sylius/Sylius-Standard/pull/781

What's left is to install and configure xdebug

diimpp reviewed (changes requested):
Xdebug increases load time for dev env drastically, so it shouldn't be added to `sylius_php_dev` stage. Idea with `docker-compose.debug.yml` looks good.

rust-le commented:
I'm closing this, if necessary, please create a separate topic
